.github/workflows/build.yaml

```diff
@@ -0,0 +1,26 @@
+name: build
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build container image
+      run: docker build . -t ghcr.io/devsisters/checkpoint:${{ github.sha }}
+    - name: Login to GitHub Container Registry
+      if: github.ref == 'refs/heads/main'
```

Comment:
Is this necessary? We already have `branches: main` on the push trigger.

Reply:
We also have `pull_request` trigger. This is necessary.

Reply:
Oh I see. How about we add [branch filters](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches) to `pull_request` instead? This seems like a more consistent pattern for limiting target branches.

Reply:
I intended to check that all pull requests can be correctly built and selectively (only in the `main` branch) pushed to GHCR.

Reply:
I understand. In this case I would have split the workflow for `push` and `pull_request` entirely, but this seems okay.